lib/r10k/git/shellgit/bare_repository.rb

```diff
@@ -24,7 +24,11 @@ def clone(remote)
     proxy = R10K::Git.get_proxy_for_remote(remote)
 
     R10K::Git.with_proxy(proxy) do
-      git ['clone', '--mirror', remote, git_dir.to_s]
+      if Gem.win_platform?
+        git ['clone', '--config', 'core.autocrlf=false', '--mirror', remote, git_dir.to_s]
```

Comment:
would autocrlf=false for all platforms work? 

I feel like r10k should never let git modify line endings of text files it's installing from what it is in the source repo, but the danger there I suppose is there may be some workflows where people are relying on this behaviour somehow (git pre-commit hook scripts that need to be executed on the windows dev box during git commit?) ... a fix in bolt may be safer, though to guard against this and make the change in r10k you could have it be a non-default option to preserve existing behaviour, eg --no-fuck-with-line-endings option that bolt can then use

Reply:
100% it would, it's what I originally had in the initial implementation. I had concerns however that this could potentially affect users running r10k on Linux. By isolating the change to Windows this update can only affect Windows systems running r10k. 

I suspect the only use case for r10k on Windows is when it used by bolt to pull modules from git repos. Unless you can think of any others? Users running r10k in workflows I suspect would be using nix only. Therefore this change would have no impact.